### PR TITLE
Add teraform module

### PR DESCRIPTION
**Pull request by misohu**
Closes: https://github.com/canonical/minio-operator/issues/186

Based on this PR: https://github.com/canonical/argo-operators/pull/198

**To test**

- Create juju controller with `kubeflow` model
- Run `terraform apply -var "app_name=minio" -var "channel=ckf-1.9/stable" -var "model_name=kubeflow" -var "resources={oci-image: 4}"`

### CODE REVIEW

File: terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Terraform module for minio
```
Comment (orfeas-k) [inline terraform/README.md:1]: Let's add a section in the README to mention compatibility

**Review by orfeas-k — COMMENTED** [review]
tested and works. left 1 comment

File: terraform/variables.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  description = "Model name"
+  type        = string
+}
+
+# TODO: Update to a map of strings, once juju provider 0.14 is released
+variable "resources" {
+  description = "Map of resources revisions"
+  type        = map(number)
+  default     = null
+}
```
Comment (orfeas-k) [inline terraform/variables.tf:29]: let's update according to https://github.com/canonical/mlmd-operator/pull/99

File: terraform/versions.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.6"
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = "~> 0.13.0"
+    }
+  }
```
Comment (orfeas-k) [inline terraform/versions.tf:8]: let's update according to https://github.com/canonical/mlmd-operator/pull/99

File: terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+### Inputs
+The module offers the following configurable inputs:
+
+| Name | Type | Description | Required |
+| - | - | - | - |
+| `app_name`| string | Application name | False |
+| `channel`| string | Channel that the charm is deployed from | False |
+| `config`| map(string) | Map of the charm configuration options | False |
+| `model_name`| string | Name of the model that the charm is deployed on | True |
+| `resources`| map(number) | Map of charm resources revisions | False |
```
Comment (orfeas-k) [inline terraform/README.md:19]: This will need to be updated too

File: terraform/variables.tf (hunk 2 above)
Comment (orfeas-k) [reply terraform/variables.tf:29]: https://github.com/canonical/mlmd-operator/pull/99/files#r1764724422

File: terraform/variables.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  default     = {}
+}
+
+variable "model_name" {
+  description = "Model name"
+  type        = string
+}
+
+# TODO: Update to a map of strings, once juju provider 0.14 is released
+variable "resources" {
```
Comment (mvlassis) [inline terraform/variables.tf:25]:
```suggestion
variable "resources" {
```

File: .gitignore [hunk 6]
```diff
@@ -5,3 +5,5 @@ build/
 __pycache__/
 .tox
 .idea
```
Comment (mvlassis) [inline .gitignore:7]: Not strictly needed, but useful to add in all repos
```suggestion
.idea
venv/
```

File: terraform/variables.tf [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+variable "revision" {
+  description = "Charm revision"
+  type        = number
+  default     = null
+}
+
+variable "storage_directives" {
+  description = "Charm storage"
```
Comment (mvlassis) [inline terraform/variables.tf:38]: To match this merged PR: https://github.com/canonical/mlmd-operator/pull/99/files
```suggestion
  description = "Map of storage directives"
```

**Review by mvlassis — COMMENTED** [review]
Great work!

**Review by orfeas-k — APPROVED** [review]

**Pull request merged**